dao/src/main/java/org/thingsboard/server/dao/sql/query/EntityKeyMapping.java

```diff
@@ -252,7 +252,11 @@ public String toLatestJoin(QueryContext ctx, EntityFilter entityFilter, EntityTy
                 .filter(StringUtils::isNotEmpty)
                 .collect(Collectors.joining(" and "));
         if (StringUtils.isNotEmpty(filterQuery)) {
-            filterQuery = " AND (" + filterQuery + ")";
+            if (entityKey.getType().equals(EntityKeyType.ATTRIBUTE)) {
+                filterQuery = "where " + filterQuery;
```

Comment:
Please provide query examples for both cases because the scope of changes is not obvious.

Reply:
before
`select *
      from attribute_kv alias2
      WHERE alias2.entity_id = '58d2ed00-4b4c-11ee-a77e-cf98592a73b3'
        AND alias2.entity_type = 'DEVICE'
        AND alias2.attribute_key = 'active'
        AND (alias2.bool_v is not null and alias2.bool_v = true)
      ORDER BY alias2.last_update_ts DESC
      limit 1;`
      
after

`select *
from (select *
      from attribute_kv alias2
      WHERE alias2.entity_id = '58d2ed00-4b4c-11ee-a77e-cf98592a73b3'
        AND alias2.entity_type = 'DEVICE'
        AND alias2.attribute_key = 'active'
      ORDER BY alias2.last_update_ts DESC
      limit 1) as alias2
where (alias2.bool_v is not null and alias2.bool_v = true);`

Reply:
the main idea is to check the value part `alias2.bool_v is not null and alias2.bool_v = true` after finding the attribute.

Reply:
please provide *complete* query example. It's not the part of it but a full query.

Reply:
before
```
select count(id)
from (select entities.*
      from (select e.id id, 'DEVICE' entity_type
            from device e
            where e.tenant_id = 'bed5b770-5788-11ee-9ecf-4deeeaa08b6a') entities
               join LATERAL (select *
                             from attribute_kv alias2
                             WHERE alias2.entity_id = entities.id
                               AND alias2.entity_type = 'DEVICE'
                               AND alias2.attribute_key = 'active'
                               AND ((alias2.bool_v is not null and alias2.bool_v = true))
                             ORDER BY alias2.last_update_ts DESC
                             limit 1) as alias2 ON true) result
```
after
```
select count(id)
from (select entities.*
      from (select e.id id, 'DEVICE' entity_type
            from device e
            where e.tenant_id = 'bed5b770-5788-11ee-9ecf-4deeeaa08b6a') entities
               join LATERAL (select *
                             from (select *
                                   from attribute_kv alias2
                                   WHERE alias2.entity_id = entities.id
                                     AND alias2.entity_type = 'DEVICE'
                                     AND alias2.attribute_key = 'active'
                                   ORDER BY alias2.last_update_ts DESC
                                   limit 1) as alias2
                             where (alias2.bool_v is not null and alias2.bool_v = true)) as alias2 ON true) result
```